app/src/main/java/de/ur/explure/views/FavoriteRoutesFragment.kt

```diff
@@ -0,0 +1,90 @@
+import de.ur.explure.viewmodel.FavoriteRoutesViewModel
+import kotlinx.android.synthetic.main.fragment_created_routes.*
+import kotlinx.android.synthetic.main.fragment_favorite_routes.*
+import kotlinx.android.synthetic.main.fragment_favorite_routes.userNameTextView
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class FavoriteRoutesFragment : Fragment(R.layout.fragment_favorite_routes),
+        RouteAdapter.OnItemClickListener, RouteAdapter.OnItemLongClickListener {
+
+    private val viewModel: FavoriteRoutesViewModel by viewModel()
+    private lateinit var adapter: RouteAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initializeAdapter()
+
+        observeUserModel()
+        observeRouteModel()
+        viewModel.getUserInfo()
+        viewModel.getFavoriteRoutes()
+    }
+
+    private fun initializeAdapter() {
+        adapter = RouteAdapter(this, this)
+        createdRoutesRecyclerView.adapter = adapter
+        createdRoutesRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        createdRoutesRecyclerView.setHasFixedSize(true)
+    }
+
+    private fun observeUserModel() {
+        viewModel.user.observe(viewLifecycleOwner, { user ->
+            if (user != null) {
+                userNameTextView.text = user.name
+            }
+        })
+    }
+
+    private fun observeRouteModel() {
+        viewModel.favoriteRoutes.observe(viewLifecycleOwner, { routes ->
+            if (routes != null) {
+                adapter.routeList = routes
+            }
+        })
+    }
+
+    override fun onItemClick(position: Int) {
+        viewModel.navigateToSinglePage()
+    }
+
+    override fun onItemLongClick(position: Int) {
+        showDialog(adapter.routeList[position])
+    }
+
+    private fun showDialog(route: Route) {
+        val dialog: AlertDialog
+        val builder = AlertDialog.Builder(this.context)
+
+        builder.setTitle(String.format(getResources().getString(R.string.remove_favorite)))
+        builder.setMessage(String.format(getResources().getString(R.string.remove_favorite_message_one)) +
+                route.title + String.format(getResources().getString(R.string.remove_favorite_message_two)))
+
+        val dialogClickListener = DialogInterface.OnClickListener { _, which ->
+            when (which) {
+                DialogInterface.BUTTON_POSITIVE -> {
+                    viewModel.removeRouteFromFavoriteRoutes(route)
+                }
+                DialogInterface.BUTTON_NEGATIVE -> {
+                    // do nothing
+                }
+            }
+        }
+
+        builder.setPositiveButton(String.format(getResources().getString(R.string.yes)), dialogClickListener)
+        builder.setNegativeButton(String.format(getResources().getString(R.string.no)), dialogClickListener)
+
+        dialog = builder.create()
+        dialog.show()
```

Comment:
Die showDialog - Methode ist etwas umständlich. Das geht in Kotlin deutlich eleganter. Z.B. so (wenn du String-Templates verwendest):
```
with(MaterialAlertDialogBuilder(requireContext())) {
    setTitle(resources.getString(R.string.remove_favorite))
    setMessage(resources.getString(R.string.remove_favorite_route_message, route.title))
    setPositiveButton(resources.getString(R.string.yes)) { _, _ -> viewModel.removeRouteFromFavoriteRoutes(route) }
    setNegativeButton(resources.getString(R.string.no)) { _, _ -> }
    show()
}
```